### ISSUE TITLE
Issue with the Unet output for multiple samples input

### ISSUE THREAD

**Issue by sinistro25** [qubvel/segmentation_models.pytorch#240]
I'm having a issue, that can be reproduced following the code: 
 
```  
from segmentation_models_pytorch import Unet
import torch as tc

net = Unet(encoder_name="resnet18")

x = tc.zeros((2, 3, 63, 63))
x[1] = 1

output = net.forward(x[:1])
output2 = net.forward(x)[:1]

print(tc.all(output == output2))
print(output)
print(output2[:1])
```

> Ouput:

```
tensor(False)
tensor([[[[ 0.5534,  0.1489, -0.3553,  ...,  0.3635,  0.1147,  0.0472],
          [-0.5823,  0.0853,  0.1402,  ...,  0.0642,  0.3019, -0.1135],
          [-0.6583,  0.4691,  0.1226,  ..., -0.1393, -0.0141, -0.2210],
          ...,
          [-0.1980,  0.0414, -0.7993,  ..., -0.2370,  0.6750,  0.0583],
          [-0.2626, -0.4830, -1.0517,  ...,  0.1245,  1.0245, -0.2520],
          [-0.3004,  0.7510,  0.5257,  ..., -0.0285, -0.1995, -0.6821]]]],
       grad_fn=<MkldnnConvolutionBackward>)
tensor([[[[ 0.6713, -0.0155,  0.0919,  ..., -0.5836,  0.2407, -0.1264],
          [-0.1182, -0.5364, -0.2901,  ...,  0.3750,  1.3623,  0.9694],
          [-0.1095, -0.1868,  0.3173,  ...,  1.2463,  1.1319,  0.5209],
          ...,
          [-0.5128, -0.1926, -0.1921,  ..., -0.5878, -0.2966, -0.1506],
          [-0.2830,  0.1687, -0.0513,  ...,  0.6669,  1.4751, -0.0419],
          [-0.2494,  0.5179,  0.3435,  ..., -0.5993, -0.6639, -0.8415]]]],
       grad_fn=<SliceBackward>)

```
The input shape is (n_samples,n_channels, height,width). First I pass a single sample (all zeros) through the net and save its output, then I pass two different samples at he same time and save the value of the output relative to the first sample (all zeros)
I would expect that the saved values should be equal, but I get  very different values.

I believe this behaviour is wrong, shouldn't the samples prediction be independent?

**Comment by qubvel** [comment]
You should make model.eval(), in other case batchnorm layers statistics will change.

**Comment by sinistro25** [comment]
Thanks, that works!! I didn't know this detail about batchnorm.
